ucsmsdk/utils/comparesyncmo.py

```diff
@@ -0,0 +1,489 @@
+                              Modify the ref object by diff object on ref ucsm
+
+                            "<=" Removes the ref obj from ref ucsm
+
+                            "==" object is equal on both ref and diff ucsm
+                        diff_property  - property list with different value
+
+    Example:
+        #1
+        ref_mos = [ref_handle.query_dn(dn="org-root/ls-sp")]
+        diff_mos = [diff_handle.query_dn(dn="org-root/ls-sp")]
+        compare_ucs_mo(ref_handle, ref_mos,
+                                   diff_handle, diff_mos)
+
+        #2
+        ref_mos = [ref_handle.query_dn(dn="org-root/org-ref/ls-sp")]
+        diff_mos = [diff_handle.query_dn(dn="org-root/org-diff/ls-sp")]
+        compare_ucs_mo(ref_handle, ref_mos,
+                                   diff_handle, diff_mos,
+                                   xlate_org = "org-root/org-ref")
+
+        #3
+        ref_mos = [ref_handle.query_dn(dn="org-root/ls-ref")]
+        diff_mos = [diff_handle.query_dn(dn="org-root/ls-diff")]
+        compare_ucs_mo(ref_handle, ref_mos, diff_handle, diff_mos,
+                        xlate_map = {"org-root/ls-diff": "org-root/ls-ref"})
+    """
+
+    reference_dict = {}
+    difference_dict = {}
+
+    if ref_obj is not None and \
+            isinstance(ref_obj, list) and len(ref_obj) > 0:
+        for mo in ref_obj:
+            if mo is None:
+                continue
+            if isinstance(mo, GenericMo):
+                if ucsgenutils.word_u(mo.get_class_id()) in skip_mos:
+                    continue
+            elif mo.get_class_id() in skip_mos or \
+                    mo.mo_meta.inp_out == MoMeta.ACCESS_TYPE_OUTPUTONLY or \
+                    (mo.mo_meta.inp_out == MoMeta.ACCESS_TYPE_IO
+                     and len(mo.mo_meta.access) == 1
+                     and mo.mo_meta.access[0] == "read-only"):
+                continue
+            reference_dict[mo.dn] = mo
+    # log.debug("reference_dict: %s" % reference_dict)
+
+    if diff_obj is not None and \
+            isinstance(diff_obj, list) and len(diff_obj) > 0:
+        for mo in diff_obj:
+            if mo is None:
+                continue
+            if isinstance(mo, GenericMo):
+                if ucsgenutils.word_u(mo.get_class_id()) in skip_mos:
+                    continue
+            elif mo.get_class_id() in skip_mos or \
+                    mo.mo_meta.inp_out == MoMeta.ACCESS_TYPE_OUTPUTONLY or \
+                    (mo.mo_meta.inp_out == MoMeta.ACCESS_TYPE_IO
+                     and len(mo.mo_meta.access) == 1
+                     and mo.mo_meta.access[0] == "read-only"):
+                continue
+            translated_mo = _translate_managed_object(mo, xlate_org, xlate_map)
+            difference_dict[translated_mo.dn] = translated_mo
+    # log.debug("difference_dict: %s" % difference_dict)
+
+    # union of reference and difference
+    dn_list = list(set.union(set(reference_dict), set(difference_dict)))
```

Comment:
@ragupta-git this code needs some changes. specifically around cases, when the `exclude_different` flag is True. In those cases, we don't really need to check for anything other than the intersection of the two dicts. Looping for every dn and checking the flag is slightly inefficient.